### PR TITLE
fix: respond to events

### PR DESCRIPTION
**Pull request by zzmp**
- Prevents re-keying web3-react when `Widget` props change
  Effectively, this was preventing state from updating in async handlers (such as event handlers).
- Simplifies event handler code throughout:
  - Integrates typings in SwapProps
  - Excludes values from hook deps (to reduce hook calls)
  - Simplifies cosmos `useEventHandler` to handle all arguments

### CODE REVIEW

File: src/components/Swap/SwapButton/index.tsx [hunk 1]
```diff
@@ -159,9 +159,8 @@ export default memo(function SwapButton({ disabled }: SwapButtonProps) {
         ? { action: approvalAction }
         : trade.state === TradeState.VALID
         ? {
-            onClick: () => {
-              const promise = onReviewSwapClick && onReviewSwapClick()
-              return promise ? promise.then((open) => setOpen(open)) : setOpen(true)
+            onClick: async () => {
+              setOpen((await onReviewSwapClick?.()) ?? true)
```
Comment (JFrankfurt) [inline src/components/Swap/SwapButton/index.tsx:163]: what happens here if onReviewSwapClick throws?

File: src/components/Swap/index.tsx [hunk 2]
```diff
@@ -45,20 +45,17 @@ function getTransactionFromMap(
...
 }
 
 export default function Swap(props: SwapProps) {
   useValidate(props)
-  useSyncConvenienceFee(props)
-  useSyncTokenDefaults(props)
-  useSyncEventHandlers(props)
+  useSyncEventHandlers(props as EventHandlers)
+  useSyncConvenienceFee(props as FeeOptions)
+  useSyncTokenDefaults(props as TokenDefaults)
```
Comment (JFrankfurt) [inline src/components/Swap/index.tsx:58]: 👍

File: src/components/TokenSelect/index.tsx [hunk 3]
```diff
@@ -133,18 +133,8 @@ export default memo(function TokenSelect({ collapsed, disabled, field, onSelect,
...
-        .then((open) => {
-          setOpen(open)
-        })
-        .catch(() => {
-          setOpen(false)
-        })
-    }
-    return setOpen(true)
+  const onOpen = useCallback(async () => {
+    setOpen((await onTokenSelectorClick?.(field)) ?? true)
```
Comment (JFrankfurt) [inline src/components/TokenSelect/index.tsx:137]: I'm pretty sure this drops the `setOpen(false)` on promise rejection functionality

File: src/components/Swap/SwapButton/index.tsx (hunk 1 above)
Comment (zzmp) [reply src/components/Swap/SwapButton/index.tsx:163]: wow good call

File: src/components/TokenSelect/index.tsx (hunk 3 above)
Comment (zzmp) [reply src/components/TokenSelect/index.tsx]: you're right, I'll back these changes out

Comment (zzmp) [reply src/components/TokenSelect/index.tsx]: this was caught by tests, too :)

**Review by JFrankfurt — APPROVED** [review]

**Pull request merged**